File: app/src/main/java/com/lab/movietime/Interface/DBHandler.java
package com.lab.movietime.Interface;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class DBHandler {

    static final String KEY_FILMID = "id_movie";
    static final String KEY_FILMNAME = "movie_name";
    static final String FILM_TABLE = "movies";
    static final String DATABASE_NAME = "TestDB";
    static final int DATABASE_VERSION = 1;

    static final String DATABASE_CREATION = "CREATE TABLE movies (id integer primary key autoincrement, id_movie text not null, movie_name text not null);";

    final Context context;
    DatabaseHelper DBHelper;
    SQLiteDatabase db;

    public DBHandler (Context ctx) {
        this.context = ctx;
        DBHelper = new DatabaseHelper(context);
    }

    private static class DatabaseHelper extends SQLiteOpenHelper {
        DatabaseHelper(Context context) {
            super(context, DATABASE_NAME, null, DATABASE_VERSION);
        }

        @Override
        public void onCreate(SQLiteDatabase db) {
            try {
                db.execSQL(DATABASE_CREATION);
            }
            catch (SQLException e) {
                e.printStackTrace();
            }
        }

        @Override
        public void onUpgrade(SQLiteDatabase db,int oldVersion,int newVersion){
            onCreate(db);
        }

    }

    public void deleteDB () {
        context.deleteDatabase(DATABASE_NAME);
    }


    public DBHandler open() throws SQLException {
        db = DBHelper.getWritableDatabase();
        return this;
    }


    public void close() {
        DBHelper.close();
    }


    public long insertMovie(String id, String name) {
        ContentValues initialValues = new ContentValues();
        initialValues.put(KEY_FILMID, id);
        initialValues.put(KEY_FILMNAME, name);
        return db.insert(FILM_TABLE, null, initialValues);
    }


    public Cursor getMovies()  {
        return db.query(FILM_TABLE, new String[] {KEY_FILMID, KEY_FILMNAME}, null, null, null, null, null);
    }

    public void removeMovie(String id) {
        String sql = "DELETE FROM movies WHERE id_movie="+id;
        db.execSQL(sql);
    }
}

File: app/src/main/java/com/lab/movietime/Interface/DBHandler.kt
package com.lab.movietime.Interface

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.SQLException
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DBHandler(val context: Context) {
    var DBHelper: DatabaseHelper
    var db: SQLiteDatabase? = null

    class DatabaseHelper internal constructor(context: Context?) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
        override fun onCreate(db: SQLiteDatabase) {
            try {
                db.execSQL(DATABASE_CREATION)
            } catch (e: SQLException) {
                e.printStackTrace()
            }
        }

        override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
            onCreate(db)
        }
    }

    fun deleteDB() {
        context.deleteDatabase(DATABASE_NAME)
    }

    @Throws(SQLException::class)
    fun open(): DBHandler {
        db = DBHelper.writableDatabase
        return this
    }

    fun close() {
        DBHelper.close()
    }

    fun insertMovie(id: String?, name: String?): Long {
        val initialValues = ContentValues()
        initialValues.put(KEY_FILMID, id)
        initialValues.put(KEY_FILMNAME, name)
        return db!!.insert(FILM_TABLE, null, initialValues)
    }

    val movies: Cursor
        get() = db!!.query(FILM_TABLE, arrayOf(KEY_FILMID, KEY_FILMNAME), null, null, null, null, null)

    fun removeMovie(id: String) {
        val sql = "DELETE FROM movies WHERE id_movie=$id"
        db!!.execSQL(sql)
    }

    companion object {
        const val KEY_FILMID = "id_movie"
        const val KEY_FILMNAME = "movie_name"
        const val FILM_TABLE = "movies"
        const val DATABASE_NAME = "TestDB"
        const val DATABASE_VERSION = 1
        const val DATABASE_CREATION = "CREATE TABLE movies (id integer primary key autoincrement, id_movie text not null, movie_name text not null);"
    }

    init {
        DBHelper = DatabaseHelper(context)
    }
}